src/components/wallet/transformRawData.tsx

```diff
@@ -0,0 +1,121 @@
+import { HistoricTransactionsQueryResult } from '../../graphql/protocol';
+import { HistoricTransaction } from './types';
+import { DataState } from '../../context/DataProvider/types';
+import { BigDecimal } from '../../web3/BigDecimal';
+import { formatUnix } from './utils';
+
+export const transformRawData = (
+  data: HistoricTransactionsQueryResult['data'],
+  dataState?: DataState,
+): HistoricTransaction[] => {
+  if (!data || !dataState) {
+    return [];
+  }
+
+  const { transactions } = data;
+  const { mAsset, bAssets } = dataState as DataState;
+
+  return transactions
+    .map(({ hash, ...tx }) => {
+      const timestamp = parseInt(tx.timestamp, 10);
+      const formattedDate = formatUnix(timestamp);
+      switch (tx.__typename) {
+        case 'RedeemTransaction': {
+          const mappedBassets = tx.bassets.map(
+            basset => bAssets[basset.id].symbol,
+          );
+          const massetUnits = new BigDecimal(tx.massetUnits, mAsset.decimals);
+          return {
+            description: `You've redeemed ${massetUnits.simple.toFixed(2)} of ${
+              mAsset.symbol
+            } with ${mappedBassets}`,
+            hash,
+            timestamp,
+          };
+        }
+        case 'RedeemMassetTransaction': {
+          const massetUnits = new BigDecimal(tx.massetUnits, mAsset.decimals);
+          return {
+            description: `You've redeemed ${massetUnits.simple.toFixed(2)} of ${
```

Comment:
Let's also change the wording here:

"You redeemed XXX mUSD proportionally into all assets"